app/pages/project/floating-ips/FloatingIpsPage.tsx

```diff
@@ -0,0 +1,258 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { Link, Outlet, type LoaderFunctionArgs } from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQueryClient,
+  usePrefetchedApiQuery,
+  type FloatingIp,
+  type Instance,
+} from '@oxide/api'
+import { useQueryTable, type MenuAction } from '@oxide/table'
+import {
+  buttonStyle,
+  EmptyMessage,
+  IpGlobal24Icon,
+  Listbox,
+  Modal,
+  Networking24Icon,
+  PageHeader,
+  PageTitle,
+  TableActions,
+} from '@oxide/ui'
+
+import { getProjectSelector, useProjectSelector } from 'app/hooks'
+import { confirmDelete } from 'app/stores/confirm-delete'
+import { addToast } from 'app/stores/toast'
+import { pb } from 'app/util/path-builder'
+
+const EmptyState = () => (
+  <EmptyMessage
+    icon={<Networking24Icon />}
+    title="No Floating IPs"
+    body="You need to create a Floating IP to be able to see it here"
+    buttonText="New Floating IP"
+    buttonTo={pb.floatingIpsNew(useProjectSelector())}
+  />
+)
+
+FloatingIpsPage.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project } = getProjectSelector(params)
+  await Promise.all([
+    apiQueryClient.prefetchQuery('floatingIpList', {
+      query: { project, limit: 25 },
+    }),
+    apiQueryClient.prefetchQuery('instanceList', {
+      query: { project },
+    }),
+  ])
+  return null
+}
+
+export function FloatingIpsPage() {
+  const [attachModalOpen, setAttachModalOpen] = useState(false)
+  const [detachModalOpen, setDetachModalOpen] = useState(false)
+  const [floatingIpToModify, setFloatingIpToModify] = useState<FloatingIp | null>(null)
+  const queryClient = useApiQueryClient()
+  const { project } = useProjectSelector()
+  const { data: instances } = usePrefetchedApiQuery('instanceList', {
+    query: { project },
+  })
+
+  const deleteFloatingIp = useApiMutation('floatingIpDelete', {
+    onSuccess() {
+      queryClient.invalidateQueries('floatingIpList')
+      addToast({ content: 'Your Floating IP has been deleted' })
+    },
+  })
+
+  const makeActions = (floatingIp: FloatingIp): MenuAction[] => {
+    const isAttachedToAnInstance = !!floatingIp.instanceId
+    return [
+      {
+        label: 'Attach',
+        disabled: isAttachedToAnInstance
+          ? 'This floating IP must be detached from the existing instance before it can be attached to a new one'
+          : false,
+        onActivate() {
+          setFloatingIpToModify(floatingIp)
+          setAttachModalOpen(true)
+        },
+      },
+      {
+        label: 'Detach',
+        disabled: isAttachedToAnInstance
+          ? false
+          : 'This floating IP is not attached to an instance',
+        onActivate() {
+          setFloatingIpToModify(floatingIp)
+          setDetachModalOpen(true)
```

Comment:
This whole confirm process can be done a bit more automatically. See IP pool unlink:

https://github.com/oxidecomputer/console/blob/d4ec192725e7410c2f5b4d653fc223eb04f0569b/app/pages/system/networking/IpPoolPage.tsx#L118-L136